### PR TITLE
ocamlPackages.janeStreet: 0.12 → 0.13

### PR DESCRIPTION
**Pull request by vbgl**
###### Motivation for this change

Support for OCaml 4.10

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/bistro/default.nix [hunk 1]
```diff
@@ -12,6 +12,11 @@ buildDunePackage rec {
     sha256 = "114gq48cpj2mvycypa9lfyqqb26wa2gkdfwkcqhnx7m6sdwv9a38";
   };
 
+  patches = [(fetchpatch {
+    url = https://github.com/pveber/bistro/commit/0931db43a146ad7829dff5120161a775f732a878.patch;
```
Comment (bcdarwin) [inline pkgs/development/ocaml-modules/bistro/default.nix:16]: Quote `url`?  Also, consider including a comment regarding this patch?

**Review by bcdarwin — CHANGES_REQUESTED** [review]
Maybe explain in the commit message why async_ssl was removed?

File: pkgs/development/ocaml-modules/bistro/default.nix (hunk 1 above)
Comment (vbgl) [reply pkgs/development/ocaml-modules/bistro/default.nix]: Done. Thanks.

**Review by bcdarwin — APPROVED** [review]

File: pkgs/development/ocaml-modules/janestreet/janePackage_0_13.nix [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{ lib, fetchFromGitHub, buildDunePackage, defaultVersion ? "0.13.0" }:
+
+{ pname, version ? defaultVersion, hash, ...}@args:
+
+buildDunePackage (args // {
+  inherit version;
+
+  minimumOCamlVersion = "4.08";
+
+  src = fetchFromGitHub {
```
Comment (sternenseemann) [inline pkgs/development/ocaml-modules/janestreet/janePackage_0_13.nix:10]: janestreet now also has dune release tarballs, i. e. `https://github.com/janestreet/${pname}/releases/download/v${version}/${pname}-v${version}.tbz`, maybe its worth it to use them via `fetchurl`?

Comment (vbgl) [reply pkgs/development/ocaml-modules/janestreet/janePackage_0_13.nix:10]: Who will update all the hashes?

Comment (sternenseemann) [reply pkgs/development/ocaml-modules/janestreet/janePackage_0_13.nix:10]: Oh right, probably not worth it...

**Pull request merged**